<launch>
  <arg name="car" default="true" />
  <arg name="pedestrian" default="false" />
  <arg name="use_gpu" default="false" />
  <arg name="sync" default="false" />

  <!-- car detection -->
  <group if="$(arg car)">
    <group ns="obj_car">

      <!-- argument list -->
      <arg name="model_file_car" default="$(find cv_tracker)/data/car_2008.xml"/>

      <!-- dpm detection -->
      <node pkg="cv_tracker" name="dpm" type="dpm_ocv">
        <remap from="/config/obj_car/dpm" to="/config/car_dpm"/>
        <param name="detection_class_name" type="str" value="car"/>
        <param name="model_file" type="str" value="$(arg model_file_car)"/>
        <param name="use_gpu" type="bool" value="$(arg use_gpu)"/>
        <remap from="/image_raw" to="/sync_drivers/image_raw" if="$(arg sync)" />
      </node>
    </group>
  </group>


  <!-- pedestrian detection -->
  <group if="$(arg pedestrian)">
    <group ns="obj_person">

      <!-- argument list -->
      <arg name="model_file_pedestrian" default="$(find cv_tracker)/data/person.xml"/>

      <!-- dpm detection -->
      <node pkg="cv_tracker" name="dpm" type="dpm_ocv">
        <remap from="/config/obj_person/dpm" to="/config/pedestrian_dpm"/>
        <param name="detection_class_name" type="str" value="person"/>
        <param name="model_file" type="str" value="$(arg model_file_pedestrian)"/>
        <param name="use_gpu" type="bool" value="$(arg use_gpu)"/>
        <remap from="/image_raw" to="/sync_drivers/image_raw" if="$(arg sync)" />
      </node>
    </group>
  </group>

</launch>
